lib/src/gutter/gutter.dart

```diff
@@ -32,39 +38,67 @@ class GutterWidget extends StatelessWidget {
   Widget _buildOnChange(BuildContext context, Widget? child) {
     final code = codeController.code;
 
-    final tableRows = [
-      for (final i in code.hiddenLineRanges.visibleLineNumbers)
-        TableRow(
-          children: [
-            Text(
-              '${i + 1}',
-              style: style.textStyle,
-              textAlign: style.textAlign,
-            ),
-            const SizedBox(),
-            const SizedBox(),
-          ],
-        ),
-    ];
-
-    _fillIssues(tableRows);
-    _fillFoldToggles(tableRows);
+    final tableRows = List.generate(
+      code.hiddenLineRanges.visibleLineNumbers.length,
+      (i) => TableRow(
+        children: [
+          SizedBox(),
+          SizedBox(),
+          SizedBox(),
+        ],
+      ),
+    );
+
+    _fillLineNumbers(tableRows);
+
+    if (style.showErrors) {
+      _fillIssues(tableRows);
+    }
+    if (style.showFoldingHandles) {
+      _fillFoldToggles(tableRows);
+    }
+
+    final gutterWidth = style.width -
+        (style.showErrors ? 0 : _issueColumnWidth) -
+        (style.showFoldingHandles ? 0 : _foldingColumnWidth);
+
+    final issueColumnWidth = style.showErrors ? _issueColumnWidth : _padding;
+    final foldingColumnWidth =
+        style.showFoldingHandles ? _foldingColumnWidth : _padding;
```

Comment:
To avoid double padding,
1. Add a single padding always.
2. Make `_foldingColumnWidth` narrower to preserve the same look if it is shown.

Reply:
Seems like its fine to leave it as is without adding any padding.